php/commands/role.php

```diff
@@ -367,6 +383,167 @@ private static function persistence_check() {
 		if ( !$wp_roles->use_db )
 			WP_CLI::error( "Role definitions are not persistent." );
 	}
+
+	private static function get_role_capability( $role_key ) {
```

Comment:
> Here I didn't found any WordPress core function to fetch the default capability of the role.

Oh, sad :(

> There is one another way without using this method, we can add the removed capability after resetting role for dry-run. But I was not sure if that's a good idea because I think in dry-run we should not do any changes in the database.

I agree.

I'm going to focus on getting v0.24.0 out the door for the next couple of days, then I'll spend a bit of time thinking about how we can get around this issue.